File: benchmarks/ruby/split_join.rb
n = (ARGV[0] || "200000").to_i
parts = []
n.times { |i| parts.push("item" + i.to_s) }
text = parts.join(",")
back = text.split(",", -1)
total = 0
back.each { |piece| total = total + piece.size }
puts "#{back.size} #{total} #{text.size}"

File: benchmarks/lua/split_join.lua
local n = tonumber(arg[1] or "200000")
local parts = {}
for i = 0, n - 1 do
    parts[#parts + 1] = "item" .. i
end
local text = table.concat(parts, ",")
local back = {}
for piece in string.gmatch(text, "([^,]*)") do
    back[#back + 1] = piece
end
local total = 0
for i = 1, #back do
    total = total + #back[i]
end
print(#back .. " " .. total .. " " .. #text)
